### PR TITLE
chore: update issue template

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour
Convert markdown issue template to form.

It allows us to use required fields to ensure we capture all relevant information and prevents users deviating form the template.

image
image
image

### Current behaviour
We use markdown forms, users can free format the template.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

### Testing instructions
You try it here
https://github.com/jamime/github-actions-public/issues/new/choose

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    validations:
+      required: true
+  - type: input
+    id: codesandbox-url
+    attributes:
+      label: CodeSandbox URL
+      description: Please fork [this template](https://codesandbox.io/s/carbon-quickstart-xi5jc), and ensure you save your changes.
+      placeholder: https://codesandbox.io/s/carbon-quickstart-xi5jc
+    validations:
+      required: true
```
Comment (DipperTheDan) [inline .github/ISSUE_TEMPLATE/bug_report.yml:39]: Do we want to make having a codesandbox example required? Sometimes users find bugs in Storybook so the codesandbox would be redundant then.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    validations:
+      required: false
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Confidentiality
+      description: >-
+        By submitting this issue, you confirm that there is no confidential or commercially sensitive information included.
+      options:
+        - label: I confirm there is no confidential or commercially sensitive information included.
```
Comment (DipperTheDan) [inline .github/ISSUE_TEMPLATE/bug_report.yml:94]: Like this little reminder too 😉

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (jamime) [reply .github/ISSUE_TEMPLATE/bug_report.yml:39]: I'll make this Codesandbox or Storybook :)

Comment (DipperTheDan) [reply .github/ISSUE_TEMPLATE/bug_report.yml:39]: Excellent. That'll be splendid 😃

**Review by chrisbarber86 — APPROVED** [review]

**Review by chrisbarber86 — APPROVED** [review]

**Review by dawidzarzycki — COMMENTED** [review]
Just an idea 💡 
What do you think about adding `reproduction steps` and make it optional? Or it is like current behavior is enough?

**Review by sjohnson-sage — APPROVED** [review]

**Comment by jamime** [comment]
> Just an idea 💡
> What do you think about adding `reproduction steps` and make it optional? Or it is like current behavior is enough?

I think the `current behaviour` should cover this.

**Pull request merged**